### DISCUSSION TITLE
Upstream prematurely closed connection while reading response header from upstream

### DISCUSSION THREAD

**Discussion by anjanaprasd** [nginx/kubernetes-ingress#6742] [General]
Hi Team,
I have deployed an OpenSearch cluster and am trying to access the service via ingress. However, I am getting an error in the NGINX ingress pod logs. The backend service is working fine, but when I access it via ingress, I receive this error. Can someone help me with this?

This is my ingress resource file

apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: opensearch-dashboards-ingress
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: /
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS" # Change to "HTTPS" if needed
    nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
    nginx.ingress.kubernetes.io/client-body-buffer-size: "8m"
    nginx.ingress.kubernetes.io/proxy-read-timeout: "3600"
    nginx.ingress.kubernetes.io/proxy-send-timeout: "3600"
    nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
spec:
  ingressClassName: nginx
  rules:
    - host: opensearch-dashboard.example.com
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: my-first-dashboards
                port:
                  number: 5601

Error message:

10.0.32.16 - - [01/Nov/2024:17:15:29 +0000] "GET / HTTP/1.1" 502 157 "-" "curl/7.81.0" "-"
2024/11/01 17:15:32 [error] 153#153: *97 upstream prematurely closed connection while reading response header from upstream, client: 

Thank you very much !

**Comment by AlexFenlon** [comment]
Hi @anjanaprasd,

The annotations you are using in your Ingress resource are processed by the [ingress-nginx](https://github.com/kubernetes/ingress-nginx) project and does not relate to the ingress controller in this [project](https://github.com/nginxinc/kubernetes-ingress)

Can you please provide the version of the ingress controller you are running? 
Either the top of your logs or the image you are using.

**Comment by anjanaprasd** [comment]
Hi @AlexFenlon 
Thank you for your prompt response. Yes, I accidentally installed a different NGINX Ingress project, but the issue is now resolved after installing the NGINX Kubernetes Ingress.
Thanks !!
